File: TG001/Models/Lote.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace TG001.Models
{
    public class Lote
    {
        [Key]
        public int ID { get; set; }

        public DateTime? DataAlteracao { get; set; }

        [Required]
        public DateTime DataGeracao { get; set; }

        [Required]
        public double ValorTotalLote { get; set; }

        [Required]
        public string StatusTransmissao { get; set; }

        [ForeignKey("Usuario")]
        public string UsuarioID { get; set; }

        public virtual Usuario Usuario { get; set; }

        public ICollection<Conta> Contas { get; set; }

    }
}

File: TG001/ClientApp/src/app/Classes/Lote.ts
import { Usuario } from "./Usuario"
import { Conta } from "./Contas/Conta"

export class Lote {
	constructor(
		public ID?: number,
		public DataAlteracao?: Date,
		public DataGeracao?: Date,
		public ValorTotalLote?: number,
		public StatusTransmissao?: string,
		public UsuarioID?: string,
		public Usuario?: Usuario,
		public Contas?: Conta[]
	) { }
}
